src/node/test/snapshot.cpp

```diff
@@ -74,14 +77,17 @@ TEST_CASE("Snapshot with merkle tree" * doctest::test_suite("snapshot"))
     ccf::MerkleTreeHistory target_tree(tree.value());
     REQUIRE(source_root_before_signature == target_tree.get_root());
 
-    target_tree.append(crypto::Sha256Hash(serialised_signature));
+    target_tree.append(ccf::entry_leaf(
+      serialised_signature, crypto::Sha256Hash::from_string("ce:0.4:")));
```

Comment:
This looked like a bug to me, but it turns out that `from_string` is _not_ `from_hex_string`. I suggest we rename this function to `of_string`, or just make it a constructor for `Sha256Hash`. So then `from_` is consistently "load an existing hash", and construction is "hash this argument(s)".

Reply:
Yes, the current digest interface isn't particularly nice, I agree. I'm tempted to go for an extra constructor, since all the other ones digest their inputs, whereas the from_* are conversions.

If I could start from scratch, I think I'd be tempted to make both named.